service/worker/scheduler/workflow.go

```diff
@@ -1002,6 +1004,39 @@ func (s *scheduler) getListInfo(inWorkflowContext bool) *schedpb.ScheduleListInf
 	}
 }
 
+func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.SearchAttributes) {
+	if searchAttributes == nil {
+		return
+	}
+	upsertMap := map[string]any{}
+	for key, valuePayload := range searchAttributes.GetIndexedFields() {
+		var value any
+		if err := payload.Decode(valuePayload, &value); err != nil {
+			s.logger.Error("error upserting search attributes of the scheule", "error", err)
+			continue
+		}
+		upsertMap[key] = value
+	}
+	currentSearchAttributes := workflow.GetTypedSearchAttributes(s.ctx).GetUntypedValues()
+	for key, currentValue := range currentSearchAttributes {
+		if searchattribute.IsReserved(key.GetName()) {
```

Comment:
I think we can filter these out in frontend and assume they're all good here

Reply:
We cannot. `workflow.GetTypedSearchAttributes` get the search attributes directly from the history events.
cc: @cretz

Reply:
I'm actually changing to use the untyped version for backwards compatibility.

Reply:
oh, I read this wrong, I thought this loop was over the new ones